src/AzureAppConfigurationImpl.ts

```diff
@@ -176,34 +176,70 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
         return {
             requestTracingEnabled: this.#requestTracingEnabled,
             initialLoadCompleted: this.#isInitialLoadCompleted,
-            appConfigOptions: this.#options
+            appConfigOptions: this.#options,
+            isFailoverRequest: this.#isFailoverRequest
         };
     }
 
-    async #loadSelectedKeyValues(): Promise<ConfigurationSetting[]> {
-        const loadedSettings: ConfigurationSetting[] = [];
+    async #executeWithFailoverPolicy(funcToExecute) {
+        const clientWrappers = await this.#clientManager.getClients();
+        if (clientWrappers.length === 0) {
+            this.#clientManager.refreshClients();
+            throw new Error("No client is available to connect to the target App Configuration store.");
+        }

```

Comment:
There are two refreshClients calls and two places to throw exception, however if you remove these lines, the behavior remains the same, except for the slight difference in error message.

Reply:
Two `refreshclients` calls for different situation. Since client has backOffTime so we may get empty client list when call `getClients`. `refreshClients` in line 187 is to discover new replicas when no available clients to send any request. The other refresh call in line 211 is also for new replica discovery but happens after all the available clients failed.

Reply:
Then, why do we need to `refreshClients` in line 210? Next time calling #executeWithFailoverPolicy will call `getClients` where it will also try to discover fallback clients?

Reply:
discussed offline